### DISCUSSION TITLE
✨ [Go] Built-in support for converting fs.FS into Directory

### DISCUSSION THREAD

**Discussion by GreenCappuccino** [dagger/dagger#12342] [Ideas]
### What are you trying to do?

`fs.FS `is a common interface for filesystems in Go. Copying files from an `FS` currently requires authors to walk through the filesystems and add the files into the Directory. (For example, as in [here](https://github.com/vito/bass/blob/6ad2902d4a9011a17efc03c92a45dea3cdb33abc/pkg/runtimes/dagger.go#L556-L579), and described [here](https://archive.docs.dagger.io/0.9/110632/embed-directories/)) A built-in conversion function can simplify this process.

### Why is this important to you?

Since this is a relatively common need, it could be worth putting into the Go SDK.

### How are you currently working around this?

Implementing the same walk operation.

**Comment by valorl** [comment]
Would it make sense to extend this issue with the inverse operation too? I.e. having an `fs.FS` compliant `dagger.FS` so that function that work with `fs.FS` could be passed a dagger Directory derivative (e.g. by calling `dir.FS()`) ? I've had that use case a few times already, when trying to separate the "dagger layer" from more generic lib logic.
